adaptived/src/effects/cgroup_setting_by_psi.c

```diff
@@ -80,6 +80,10 @@ int cgroup_setting_psi_init(struct adaptived_effect * const eff, struct json_obj
 	if (ret)
 		goto error;
 
+	ret = adaptived_file_exists(cgroup_path_str);
```

Comment:
The `cgroup_setting_by_psi` effect can operate on a hierarchy, thus `*` is valid in the path (and it may not map to a valid path.)

I think a better option here would be to do something like:

```
path_len = strlen(cgroup_path_str);
if (cgroup_path_str[path_len - 1] == '*') {
    /*
     * Skip file existence check because a wildcard path was provided
     */
} else {
    ret = adaptived_file_exists(cgroup_path_str);
    if (ret)
        goto error;
}

Reply:
Since there could be some copy-pasta here, you could incorporate this into `adaptived_file_exists()` with a parameter like `skip_wildcard` or something like that